### PR TITLE
Afform - Allow submissions to be disabled and limited

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Adds FormBuilder settings to disable submissions or limit the number of submissions.

image

Before
----------------------------------------
Form submissions cannot be disabled or limited.

After
----------------------------------------
image

image

### CODE REVIEW

File: ext/afform/admin/ang/afAdmin/afAdminList.controller.js [hunk 1]
```diff
@@ -32,22 +32,13 @@
       if (afform['contact_summary:label']) {
         afform.placement.push(afform['contact_summary:label']);
       }
+      if (afform.submission_date) {
```
Comment (kurund) [inline ext/afform/admin/ang/afAdmin/afAdminList.controller.js:35]: good to remove code, same reason as other comment

File: ext/afform/core/Civi/Api4/Afform.php [hunk 2]
```diff
@@ -214,11 +245,37 @@ public static function getFields($checkPermissions = TRUE) {
...
+        ];
+        $fields[] = [
+          'name' => 'submission_count',
+          'type' => 'Extra',
+          'data_type' => 'Integer',
+          'input_type' => 'Number',
+          'description' => 'Number of submission records for this form',
+          'readonly' => TRUE,
+        ];
+        $fields[] = [
```
Comment (kurund) [inline ext/afform/core/Civi/Api4/Afform.php:265]: good to remove this code, same reason as other comment

File: ext/afform/admin/ang/afGuiEditor/config-form.html [hunk 3]
```diff
@@ -114,25 +114,46 @@
...
+        <label>
+          <input type="checkbox" ng-model="editor.afform.create_submission" >
+          {{:: ts('Log Submissions') }}
+        </label>
+        <p class="help-block">{{:: ts('Keep a log of the date, time, user, and items saved by each form submission.') }}</p>
+      </div>
+
+      <div class="form-inline" ng-if="editor.afform.create_submission">
+        <label for="submit_limit">{{:: ts('Maximum Submissions') }}</label>
+        <input type="number" min="1" step="1" id="submit_limit" ng-model="editor.afform.submit_limit">
```
Comment (kurund) [inline ext/afform/admin/ang/afGuiEditor/config-form.html:144]: We should add help text here explaining that if Max is not specified than there is no limit.

**Review by kurund — COMMENTED** [review]
Besides specific comments everything else looks good to me and working as expected!

File: ext/afform/admin/ang/afGuiEditor/config-form.html (hunk 3 above)
Comment (colemanw) [reply ext/afform/admin/ang/afGuiEditor/config-form.html]: @kurund that's a good idea. I added a placeholder:
image

File: ext/afform/admin/ang/afAdmin.js [hunk 4]
```diff
@@ -11,7 +11,7 @@
           // Load data for lists
           afforms: function(crmApi4) {
             return crmApi4('Afform', 'get', {
-              select: ['name', 'title', 'type', 'server_route', 'is_public', 'has_local', 'has_base', 'base_module', 'base_module:label', 'is_dashlet', 'contact_summary:label']
+              select: ['name', 'title', 'type', 'server_route', 'is_public', 'submission_count', 'submission_date', 'submit_limit', 'submit_enabled', 'submit_currently_open', 'has_local', 'has_base', 'base_module', 'base_module:label', 'is_dashlet', 'contact_summary:label']
```
[the remark this answers is not in the document]
Comment (kurund) [reply ext/afform/admin/ang/afAdmin.js:14]: Oh, now I see the change.

**Review by kurund — APPROVED** [review]
Looks good. Ready to merge on test pass.

**Comment by colemanw** [comment]
retest this please

**Comment by colemanw** [comment]
Jenkins is having a bad day. retest this please

**Pull request merged**